src/core/nixl_agent.cpp

```diff
@@ -281,6 +281,16 @@ nixlAgent::createBackend(const nixl_backend_t &type,
     return NIXL_ERR_BACKEND;
 }
 
+nixl_status_t
+nixlAgent::queryMem(const nixl_reg_dlist_t &descs,
+                    std::vector<nixl_query_resp_t> *&resp,
+                    const nixl_opt_args_t *extra_params) {
+    if ((extra_params->backends.size() == 0) || (extra_params->backends.size() > 1))
```

Comment:
maybe
```suggestion
    if (extra_params->backends.size() != 1)
```
btw why can't we pass a backend explicitly?

Reply:
yes, this is better. I explained in a previous comment that Yossi suggested not to break our current API style, as sooner than later we would replace it with method overloading anyways.

Reply:
fixed.